src/sagemaker_training/process.py

```diff
@@ -40,6 +40,29 @@
 _DEFAULT_BUF_SIZE = 1024 * 64
 
 
+def get_debugger_exception_classes():
+    """Set exception classes"""
+    exception_classes = []
+    if os.environ.get("USE_SMDEBUG") == "0":
+        logger.info(
+            "sagemaker debugger is not enabled, using exception types without debugger ones."
+        )
+    else:
+        try:
+            from smdebug import exceptions
+
+            # list of exceptions debugger wants training toolkit to catch and log
+            exception_classes += [ex for ex in dir(exceptions) if isclass(getattr(exceptions, ex))]
+        except ImportError:
+            logger.info(
+                "sagemaker debugger is not installed, using exception types without debugger ones."
+            )
+
+    if not exception_classes:
+        exception_classes = [errors.ExecuteUserScriptError]
```

Comment:
Let's move this logic a bit?
At the start of the file, include `DEFAULT_ERROR_CLASS = errors.ExecuteUserScriptError` and change this to 
`exception_classes = [DEFAULT_ERROR_CLASS]`. This will be more clear to understand what is the default and how we fall back if no other classes are present.

Reply:
edited.